main.go

```diff
@@ -156,13 +156,32 @@ func main() {
 		"chs": {"Change signal of x", 1, false, func(a []float64) (float64, error) { return a[0] * -1, nil }},
 		"inv": {"Invert x (1/x)", 1, false, func(a []float64) (float64, error) { return 1 / a[0], nil }},
 		"^":   {"Raise y to the power of x", 2, false, func(a []float64) (float64, error) { return math.Pow(a[0], a[1]), nil }},
-		"%":   {"Calculates y modulo x", 2, false, func(a []float64) (float64, error) { return math.Mod(a[0], a[1]), nil }},
-		"pct": {"Calculate x% of y", 2, false, func(a []float64) (float64, error) { return a[0] * a[1] / 100, nil }},
+		"mod":   {"Calculates y modulo x", 2, false, func(a []float64) (float64, error) { return math.Mod(a[0], a[1]), nil }},
+		"%": {"Calculate x% of y", 2, false, func(a []float64) (float64, error) { return a[0] * a[1] / 100, nil }},
+
+        "fac": {"Calculate factorial of x", 1, false, func(a []float64) (float64, error) {
+            var x = uint64(a[0])
+            if (x <= 0) {
+                return 0, errors.New("fact requires a positive number")
+            }
+            var fact = uint64(1)
+            for ix := uint64(1); ix <= x; ix++ {
+			    fact *= ix
+		    }
+            return float64(fact), nil
+		  }},
 
 		// stack operations
 		"p": {"Display stack", 0, true, func(_ []float64) (float64, error) { stack.print(); return 0, nil }},
-		"c": {"Clear stack", 0, true, func(_ []float64) (float64, error) { stack.clear(); return 0, nil }},
+		"clr": {"Clear stack", 0, true, func(_ []float64) (float64, error) { stack.clear(); return 0, nil }},
 		"=": {"Print top of stack (x)", 0, true, func(_ []float64) (float64, error) { fmt.Println(stack.top()); return 0, nil }},
+		"d": {"Drop top of stack (x)", 1, true, func(_ []float64) (float64, error) { return 0, nil }},
+
+        // math & physical constants
+        "pi":  {"The famous transcedental number", 0, false, func(_ []float64) (float64, error)    { return 3.1415926536, nil }},
```

Comment:
Constants should be prefixed by something or uppercase. Maybe const.pi? or PI?
Also, instead of using the number, use the constant from math. For example, Math.Pi, Math.E, etc (https://cs.opensource.google/go/go/+/refs/tags/go1.23.1:src/math/const.go;l=11)

Reply:
Done, all are now uppercase; also all use golang libraries except for "mol" because, AFAICS, no golanf library defines it.